nbgrader/converters/base.py

```diff
@@ -43,9 +43,21 @@ class BaseConverter(LoggingConfigurable):
         )
     ).tag(config=True)
 
+    groupshared = Bool(
+        False,
+        help=dedent(
+            """
+            Be less strict about user permissions (instructor files are by
+            default group writeable.  Requires that admins ensure that primary
+            groups are correct!
+            """
+        )
+    ).tag(config=True)
+
     @default("permissions")
     def _permissions_default(self):
-        return 444
+        return 664 if self.groupshared else 644
```

Comment:
So I think that the permissions should still be 444 when running the autograder, which is what this line does (it looks like for `assign` and `feedback`, it gets overwritten to 644). The reasoning behind that is to prevent instructors from modifying the autograded notebooks (mostly to prevent confusion, so students don't get back a notebook that is different from the one they submitted). So I'm fine changing the default here but we should also change it to 444 in the autograder then.

Reply:
I think one consideration was: if instructor1 autogrades, then it creates some files which instructor2 can't remove.  So no one else can autograde an updated assignment.  (This was a problem somewhere, but I may be remembering something wrong).

Reply:
Oh, hmm, I see. I think though that if the file has the right group ownership, then other people in the group should be able to remove the file, even if write permissions are disabled. At least, you do see a similar issue with single instructors---they can't directly remove the file but because they own it they can change the permissions and then remove (and I believe this happens in nbgrader somewhere under the hood if you use --force). So I would expect you could do something similar with group-level permissions, though I haven't tested it.

Reply:
I'll have to try without to see what the problem was.  It could relate to subdirs (if you can write dir, you can remove file, but if you can't write subdir, you can't delete stuff within to get an empty dir to remove)...

I'll note to come back to this.